ff/src/fields/models/fp3.rs

```diff
@@ -42,11 +42,11 @@ impl<P: Fp3Config> CubicExtConfig for Fp3ConfigWrapper<P> {
     const NONRESIDUE: Self::BaseField = P::NONRESIDUE;
 
     const SQRT_PRECOMP: Option<SqrtPrecomputation<CubicExtField<Self>>> =
-        Some(SqrtPrecomputation::TonelliShanks(
-            P::TWO_ADICITY,
-            &P::TRACE_MINUS_ONE_DIV_TWO,
-            P::QUADRATIC_NONRESIDUE_TO_T,
-        ));
+        Some(SqrtPrecomputation::TonelliShanks {
+            two_adicity: P::TWO_ADICITY,
+            trace_of_modulus_minus_one_div_two: &P::TRACE_MINUS_ONE_DIV_TWO,
```

Comment:
the order of the parameters doesn't look right - it seems you changed it in the above definitions but not here.

Reply:
Good catch! Fixed in the latest commit.